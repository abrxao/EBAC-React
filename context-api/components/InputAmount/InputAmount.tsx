import { Input } from "@chakra-ui/react";
import { FunctionComponent, useEffect } from "react";
import { useCounter } from "../Context/Context";
import IMask from "imask";

const InputAmount: FunctionComponent = ()=>{
    const amountValue = useCounter();
    useEffect(() => {
      const maskOptions = {
        mask: Number,
        isContentEditable: true,
      };
      const amount: HTMLElement | null = document.querySelector("#amount");
      amount ? IMask(amount, maskOptions) : () => {};
    }, []);

    if(!amountValue){
        return null
    }
    const {value, changeValue} = amountValue;

    return (
      <>
        <Input maxLength={30} id="amount" onChange={e=>changeValue(e.target.value)}/>
      </>
    );
}

export default InputAmount;