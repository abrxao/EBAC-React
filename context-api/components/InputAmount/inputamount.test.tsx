import React from "react";

import { act, fireEvent, render, screen } from "@testing-library/react";
import ContextProvider from "../Context/Context";
import InputAmount from "./InputAmount";

describe("InputAmount mask tests", () => {
  const { container } = render(
    <ContextProvider>
      <InputAmount
        maxLength={30}
        id="amount-input"
        data-testid="testid"
        type="number"
      />
    </ContextProvider>
  );
  const input = screen.queryByTestId("testid") as HTMLInputElement;

  test("InputMask Snapshot", async () => {
    expect(container).toMatchSnapshot();
  });

  test("should clear the input value when non-numeric characters are entered", async () => {
    await act(async () => {
      fireEvent.keyPress(input, { target: { value: "aaaa" } });
    });
    expect(input.value).toEqual("");
  });

  test("should allow numeric characters in the input", async () => {
    await act(async () => {
      fireEvent.keyPress(input, { target: { value: "123" } }); //change(input,{ target: { value: "asaas" } } );
    });
    expect(input.value).toEqual("123");
  });
});
