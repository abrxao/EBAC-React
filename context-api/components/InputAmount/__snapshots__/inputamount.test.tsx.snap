// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAmount mask tests InputMask Snapshot 1`] = `
<div>
  <input
    class="chakra-input css-0"
    data-testid="testid"
    id="amount-input"
    maxlength="30"
    type="number"
  />
  <button
    class="chakra-button css-1ner6ak"
    data-testid="send-button"
    type="button"
  >
    convert
  </button>
</div>
`;
